File: 0724.find_pivot_index/solution.py
from typing import List


class Solution:
    def pivot_index(self, nums: List[int]) -> int:
        total_sum = sum(nums)
        this_sum = 0
        for i in range(len(nums)):
            if total_sum - nums[i] == this_sum:
                return i
            this_sum += nums[i]
            total_sum -= nums[i]
        return -1


if __name__ == '__main__':
    assert Solution().pivot_index([-1, -1, 0, 1, 1, 0]) == 5

File: 0724.find_pivot_index/solution.go
package main


func pivotIndex(nums []int) int {
	var totalSum int
	for _, v := range nums {totalSum += v}
	partSum := 0
	for i := 0; i < len(nums); i++ {
		if totalSum-nums[i] == partSum {
			return i
		}
		totalSum -= nums[i]
		partSum += nums[i]
	}
	return -1
}
